builtin/providers/aws/resource_aws_wafregional_byte_match_set.go

```diff
@@ -0,0 +1,197 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegionalByteMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegionalByteMatchSetCreate,
+		Read:   resourceAwsWafRegionalByteMatchSetRead,
+		Update: resourceAwsWafRegionalByteMatchSetUpdate,
+		Delete: resourceAwsWafRegionalByteMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"byte_match_tuples": &schema.Schema{
```

Comment:
Given how the sets and lists are represented in HCL, i.e.

```hcl
byte_match_tuples {
...
}

byte_match_tuples {
...
}
```
it would make more sense if this field name was singular - i.e. `byte_match_tuple`. I will have a look into deprecating `byte_match_tuples` in favour of singular name in WAF resources too as we seemed to overlook this in the review of that resource code there.